src/tests/functional/shared_test_classes/include/shared_test_classes/base/benchmark.hpp

```diff
@@ -0,0 +1,325 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cctype>
+#include <iostream>
+
+#include <nlohmann/json.hpp>
+#include "layer_test_utils.hpp"
+#include "ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+
+using json = nlohmann::json;
+
+class BenchmarkLayerTestReporter {
+public:
+    static constexpr const char* benchmarkReportFileName = "benchmark_layers.json";
+
+    explicit BenchmarkLayerTestReporter(bool is_readonly)
+        : is_readonly_{is_readonly} {
+        if (is_readonly) {
+            try {
+                report_file_.open(benchmarkReportFileName, std::ios::in);
+                report_json_ = json::parse(report_file_);
+            } catch (...) {
+                report_json_ = json{};
+            }
+        } else {
+            try {
+                report_file_.open(benchmarkReportFileName, std::ios::in | std::ios::out);
+                report_json_ = json::parse(report_file_);
+            } catch (...) {
+                report_file_.open(benchmarkReportFileName, std::ios::out | std::ios::trunc);
+                report_json_ = json{};
+            }
+        }
+    }
+
+    ~BenchmarkLayerTestReporter() {
+        if (!is_readonly_) {
+            report_file_ << report_json_.dump(4);
+        }
+    }
+
+    void report(const std::string& nodeTypeName,
+                const std::string& testCaseName,
+                const uint64_t time) {
+        report_json_[nodeTypeName][testCaseName] = time;
+    }
+
+    uint64_t get_time(const std::string& nodeTypeName,
+                      const std::string& testCaseName) {
+        return report_json_[nodeTypeName][testCaseName];
+    }
+
+private:
+    bool is_readonly_{};
+    std::fstream report_file_{};
+    json report_json_;
+};
+
+}  // namespace test
+}  // namespace ov
+
+namespace LayerTestsDefinitions {
+
+template <typename BaseLayerTest>
+class BenchmarkLayerTest : public BaseLayerTest {
+    static_assert(std::is_base_of<LayerTestsUtils::LayerTestsCommon, BaseLayerTest>::value,
```

Comment:
Could it be affected in the release build? I am not sure that it works as for debug..

Reply:
This is a compile-time assertion, so it does not depend on build mode. I believe @lohika-denis-kotov added this to ensure no-one would try thoughtlessly change the base class here in the future. It's like embedding comment "I want this base class constraints for the reason, otherwise it wouldn't work" directly into code.